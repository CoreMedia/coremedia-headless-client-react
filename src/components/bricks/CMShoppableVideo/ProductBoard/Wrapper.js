// @flow
// import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import theme from 'styled-theming';

import { media } from '../../../../styles/themes/utils';

const borderColor = theme('name', {
  default: 'rgba(0, 108, 174, 1)',
  hkm: '#fabfc0',
});

const fadeIn = keyframes`
  from {
    opacity 0;
  }

  to {
    opacity 1;
  }
`;

const ProductBoardWrapper = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: #fff;
  overflow-x: scroll;
  overflow-y: hidden;
  direction: rtl;
  border-width: 3px 0;
  border-style: solid none;
  border-color: ${borderColor};
  padding: 7px;
  animation: ${fadeIn} 0.6s linear;
  ${props => media(props.theme.breakpoints.tablet)`
    display: initial;
    overflow-x: hidden;
    overflow-y: scroll;
    direction: ltr;
  `};
`;

ProductBoardWrapper.displayName = 'ProductBoardWrapper';

/*ProductBoardWrapper.propTypes = {
  theme: PropTypes.shape({
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
    name: PropTypes.string.isRequired,
  }),
};*/

ProductBoardWrapper.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
    },
    name: 'default',
  },
};

export default ProductBoardWrapper;
