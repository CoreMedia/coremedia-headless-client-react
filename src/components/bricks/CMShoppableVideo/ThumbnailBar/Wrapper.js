// @flow
import styled from 'styled-components';

const ThumbnailsWrapper = styled.div`
  overflow: hidden;
`;

ThumbnailsWrapper.displayName = 'ThumbnailsWrapper';

export default ThumbnailsWrapper;
