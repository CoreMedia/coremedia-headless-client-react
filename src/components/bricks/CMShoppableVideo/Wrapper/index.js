// @flow
import styled from 'styled-components';

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  line-height: 0;
  transition: all 0.6s linear 0s;
`;

Wrapper.displayName = 'Wrapper';

export default Wrapper;
