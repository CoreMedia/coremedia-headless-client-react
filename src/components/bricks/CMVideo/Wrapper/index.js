// @flow
import type { ComponentType } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

type Props = {
  width?: number,
  height?: number,
};

const Wrapper: ComponentType<Props> = styled.div`
  margin: 0;
  padding: 0;
  cursor: pointer;
  width: ${props => (props.width ? `${props.width}px` : '100%')};
  height: ${props => (props.height ? `${props.height}px` : '100%')};
`;
Wrapper.displayName = 'Wrapper';

Wrapper.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
};

export default Wrapper;
