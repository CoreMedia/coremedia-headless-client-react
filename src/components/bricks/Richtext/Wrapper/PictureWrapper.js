// @flow
import styled from 'styled-components';

import { media } from '../../../../styles/themes/utils';

const PictureWrapper = styled.span`
  margin: 1em 0em;
  display: block;
  ${props => media(props.theme.breakpoints.tablet)`
    margin: 1em 1em 1em 0em;
    width: 50%;
    float: left;
  `};
`;

PictureWrapper.displayName = 'PictureWrapper';

/*PictureWrapper.propTypes = {
  theme: PropTypes.shape({
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
    name: PropTypes.string.isRequired
  }),
};*/

PictureWrapper.defaultProps = {
  theme: {
    name: 'default',
    breakpoints: {
      tablet: 768,
    },
  },
};

export default PictureWrapper;
