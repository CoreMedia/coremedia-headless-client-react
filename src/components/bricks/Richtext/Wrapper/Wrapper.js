// @flow
import { P } from '../../../basic/Paragraph';

const Wrapper = P.withComponent('div').extend`
  margin-top: 2em;
  padding: 0 2em;
`;
Wrapper.displayName = 'Wrapper';

export default Wrapper;
