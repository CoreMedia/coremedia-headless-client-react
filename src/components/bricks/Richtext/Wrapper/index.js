import Wrapper from './Wrapper';
import PictureWrapper from './PictureWrapper';

export { Wrapper, PictureWrapper };
