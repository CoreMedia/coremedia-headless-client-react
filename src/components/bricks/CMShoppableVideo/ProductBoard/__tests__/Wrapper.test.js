// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Wrapper from '../Wrapper';

describe('Wrapper Component', () => {
  describe('with default theme', () => {
    it('should render correctly', () => {
      const tree = renderer.create(<Wrapper>Content</Wrapper>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('with hkm theme', () => {
    it('should render correctly', () => {
      const tree = renderer
        .create(
          <Wrapper
            theme={{
              breakpoints: {
                tablet: 768,
              },
              name: 'hkm',
            }}
          >
            Content
          </Wrapper>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
