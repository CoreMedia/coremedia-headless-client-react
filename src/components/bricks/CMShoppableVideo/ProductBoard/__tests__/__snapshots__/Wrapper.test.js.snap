// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Component with default theme should render correctly 1`] = `
.c0 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: #fff;
  overflow-x: scroll;
  overflow-y: hidden;
  direction: rtl;
  border-width: 3px 0;
  border-style: solid none;
  border-color: rgba(0,108,174,1);
  padding: 7px;
  -webkit-animation: kTETKv 0.6s linear;
  animation: kTETKv 0.6s linear;
}

@media (min-width:48em) {
  .c0 {
    display: initial;
    overflow-x: hidden;
    overflow-y: scroll;
    direction: ltr;
  }
}

<div
  className="c0"
>
  Content
</div>
`;

exports[`Wrapper Component with hkm theme should render correctly 1`] = `
.c0 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: #fff;
  overflow-x: scroll;
  overflow-y: hidden;
  direction: rtl;
  border-width: 3px 0;
  border-style: solid none;
  border-color: #fabfc0;
  padding: 7px;
  -webkit-animation: kTETKv 0.6s linear;
  animation: kTETKv 0.6s linear;
}

@media (min-width:48em) {
  .c0 {
    display: initial;
    overflow-x: hidden;
    overflow-y: scroll;
    direction: ltr;
  }
}

<div
  className="c0"
>
  Content
</div>
`;
